sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProto.java

```diff
@@ -1147,10 +1146,24 @@ public static TableRow tableRowFromMessage(
       Object fieldValue = field.getValue();
       if ((includeCdcColumns || !StorageApiCDC.COLUMNS.contains(fullName.toString()))
           && includeField.test(fieldName)) {
-        tableRow.put(
-            fieldName,
+        Object convertedValue =
             jsonValueFromMessageValue(
-                fieldDescriptor, fieldValue, true, includeField, fullName.append(".").toString()));
+                fieldDescriptor, fieldValue, true, includeField, fullName.append(".").toString());
+
+        // Use setF when field name is "f" to avoid IllegalArgumentException with internal field
+        if ("f".equals(fieldName)) {
+          if (convertedValue instanceof List) {
+            @SuppressWarnings("unchecked")
+            List<TableCell> tableCells = (List<TableCell>) convertedValue;
+            tableRow.setF(tableCells);
```

Comment:
The constructed TableRow object is not quite right. fieldname ("f") is not used in two branches. From spec I understand setF is meant to sets the entire list of fields and their values at once. But here it is understood as a method to set a field named "f" (not correct).